src/NHibernate/Multi/QueryBatch.cs

```diff
@@ -44,18 +47,17 @@ public void Execute()
 			{
 				Init();
 
-				if (!Session.Factory.ConnectionProvider.Driver.SupportsMultipleQueries)
+				using (Session.BeginProcess())
```

Comment:
Unrelated slight improvement here: better have the `BeginProcess` around the lack of multi-queries support case too, in order to avoid starting processing on each query.